### PR TITLE
Fix RHMI patching

### PR DESCRIPTION
**Pull request by rnetser**
Add missing `.exists`

### CODE REVIEW

File: ocm_python_wrapper/cluster.py [hunk 1]
```diff
@@ -375,13 +375,15 @@ def _wait_for_rhmi_resource():
...
+                func=lambda: RHMI(
+                    name="rhoam", namespace="redhat-rhoam-operator"
+                ).exists,
                 exceptions_dict=NOT_FOUND_ERROR_EXCEPTION_DICT,
             ):
                 if sample:
                     return sample
 
-        rhmi = _wait_for_rhmi_resource()
+        rhmi = RHMI(name="rhoam", namespace="redhat-rhoam-operator")
```
[the remark this answers is not in the document]
Comment (myakove) [reply ocm_python_wrapper/cluster.py:387]: please define it before the forloop and reuse it.

**Review by myakove — CHANGES_REQUESTED** [review]

File: ocm_python_wrapper/cluster.py [hunk 2]
```diff
@@ -375,13 +375,16 @@ def _wait_for_rhmi_resource():
...
                 sleep=SLEEP_1SEC,
-                func=lambda: RHMI(name="rhoam", namespace="redhat-rhoam-operator"),
+                func=lambda: RHMI(
+                    name="rhoam", namespace="redhat-rhoam-operator"
+                ).exists,
                 exceptions_dict=NOT_FOUND_ERROR_EXCEPTION_DICT,
             ):
                 if sample:
-                    return sample
+                    break
```
Comment (rnetser) [inline ocm_python_wrapper/cluster.py:384]:
```
                func=lambda: RHMI(
                    name="rhoam", namespace="redhat-rhoam-operator"
                ),
                exceptions_dict=NOT_FOUND_ERROR_EXCEPTION_DICT,
            ):
                if sample and sample.exists:
                    return sample
```

and restore the original code

**Review by myakove — APPROVED** [review]

File: ocm_python_wrapper/cluster.py [hunk 3]
```diff
@@ -372,14 +372,14 @@ def uninstall_addon(self, wait=True, wait_timeout=TIMEOUT_30MIN):
     @staticmethod
     def update_rhoam_cluster_storage_config():
         def _wait_for_rhmi_resource():
-            for sample in TimeoutSampler(
+            for rhmi_sample in TimeoutSampler(
                 wait_timeout=TIMEOUT_30MIN,
                 sleep=SLEEP_1SEC,
                 func=lambda: RHMI(name="rhoam", namespace="redhat-rhoam-operator"),
```
[the remark this answers is not in the document]
Comment (rnetser) [reply ocm_python_wrapper/cluster.py]: done

**Review by myakove — CHANGES_REQUESTED** [review]
:)

**Review by myakove — APPROVED** [review]

File: ocm_python_wrapper/cluster.py [hunk 4]
```diff
@@ -372,14 +372,15 @@ def uninstall_addon(self, wait=True, wait_timeout=TIMEOUT_30MIN):
     @staticmethod
     def update_rhoam_cluster_storage_config():
         def _wait_for_rhmi_resource():
-            for sample in TimeoutSampler(
+            _rhmi = RHMI(name="rhoam", namespace="redhat-rhoam-operator")
```
Comment (oharan2) [inline ocm_python_wrapper/cluster.py:375]: Patching now works, but this line fails with non-existence (trying to reach too early before resource created):
`ocp_resources.resource WARNING Couldn't find RHMI in integreatly.org api group`
`*** NotImplementedError: Couldn't find RHMI in integreatly.org api group`

**Review by myakove — APPROVED** [review]

**Pull request merged**